src/main/java/gov/nasa/pds/tools/label/LocationValidator.java

```diff
@@ -161,130 +161,144 @@ public void validate(URL target) {
+                ArrayList<Target> ignoreBundleList = AggregateManager.buildBundleIgnoreList(url,
+                    this.labelExtension);
+                ignoreList.addAll(ignoreBundleList);
+                Target latestBundle = AggregateManager.getLatestBundle();
+
+                // Only build collection ignore list if latestBundle is not null. The reason is
+                // a bundle
+                // contains collection and if there is no bundle, then there is no collection
+                // information to be gathered.
+                if (latestBundle != null) {
+                    ArrayList<Target> ignoreCollectionList = AggregateManager.buildCollectionIgnoreList(url,
+                        latestBundle.getUrl(), this.labelExtension, true);
+                    ignoreList.addAll(ignoreCollectionList);
+                    LOG.debug("url,ignoreCollectionList {},{}", url, ignoreCollectionList);
+                    LOG.debug("url,ignoreCollectionList.size() {},{}", url, ignoreCollectionList.size());
+                }
+                LOG.debug("url,latestBundle {},{}", url, latestBundle);
+                LOG.debug("url,ignoreBundleList {},{}", url, ignoreBundleList);
+                LOG.debug("url,ignoreBundleList.size() {},{}", url, ignoreBundleList.size());
+            } else {
+                LOG.debug("Input url is a file, will not crawl for bundle/collection files {}", url);
+            }
         }
-      }
 
-      if (!rule.isApplicable(location)) {
-        LOG.error("'{}' validation style is not applicable for location {}", rule.getCaption(),
-            location);
-        return;
-      }
+        if (!rule.isApplicable(location)) {
+            LOG.error("'{}' validation style is not applicable for location {}", rule.getCaption(),
+                location);
+            return;
+        }
+
+        ValidationTask task = new ValidationTask(listener, ruleContext, targetRegistrar);
+        task.setLocation(location);
+        task.setRule(rule);
+        task.setRuleManager(ruleManager);
+        Crawler crawler = CrawlerFactory.newInstance(url);
+        // Set filter so the crawler will ignore other bundle/collection files that are
+        // not latest.
+        crawler.addAllIgnoreItems(ignoreList);
+
+        ruleContext.setCrawler(crawler);
+        ruleContext.setRule(rule);
+
+        LOG.debug("validate:Submitting task to taskManager location {} rule {} ", location,
+            rule.getCaption());
+
+        // 🟢 Progress Tracking Implementation
+        int fileCounter = 0;
+        int trackProgress = ruleContext.getEveryN(); // Get user-defined tracking interval
+
+        for (Target targetFile : crawler.crawl()) { // Loop over validated files
+            taskManager.submit(task);
+            fileCounter++;
+
+            // Log progress every N files
+            if (fileCounter % trackProgress == 0) {
+                System.out.println("Processed " + fileCounter + " files so far...");
```

Comment:
Other users are complaining that material is output to the console rather than the log file. Mostly because they ignore the console output. When the new progressPageSize is added to the CLI argument list, its default value should be 0 meaning this message will not be output: `if (trackProgress > 0 && fileCounter % trackProgress == 0)`.